src/backend/distributed/utils/connection_management.c

```diff
@@ -0,0 +1,635 @@
+/*-------------------------------------------------------------------------
+ *
+ * connection_management.c
```

Comment:
Minor: I think our utils directory is a little cramped, might be nice to put this into a connection directory.